src/main/java/com/gotocompany/depot/maxcompute/schema/SchemaDifferenceUtils.java

```diff
@@ -0,0 +1,88 @@
+package com.gotocompany.depot.maxcompute.schema;
+
+import com.aliyun.odps.TableSchema;
+import com.aliyun.odps.type.ArrayTypeInfo;
+import com.aliyun.odps.type.StructTypeInfo;
+import com.aliyun.odps.type.TypeInfo;
+import com.gotocompany.depot.maxcompute.model.MaxComputeColumnDetail;
+import org.apache.commons.lang.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import static com.gotocompany.depot.maxcompute.util.TypeInfoUtils.isPrimitiveArrayType;
+import static com.gotocompany.depot.maxcompute.util.TypeInfoUtils.isPrimitiveType;
+import static com.gotocompany.depot.maxcompute.util.TypeInfoUtils.isStructArrayType;
+import static com.gotocompany.depot.maxcompute.util.TypeInfoUtils.isStructType;
+
+public class SchemaDifferenceUtils {
```

Comment:
add javadoc, and mark as deprecated, until ali's team provide the proper table update

Reply:
the behavior of SDK update API is yet to be shared by Ali's team